src/Analyzers/CSharp/CodeFixes/UseUtf8StringLiteral/UseUtf8StringLiteralCodeFixProvider.cs

```diff
@@ -48,12 +48,26 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
+            var readOnlySpanType = semanticModel.Compilation.GetBestTypeByMetadataName(typeof(ReadOnlySpan<>).FullName!);
+            // The analyzer wouldn't raise a diagnostic if this were null
+            Contract.ThrowIfNull(readOnlySpanType);
+
             foreach (var diagnostic in diagnostics)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
                 var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
-                var stringValue = GetUtf8StringValueForDiagnostic(semanticModel, diagnostic, cancellationToken);
+                var arrayOp = GetArrayCreationOperation(semanticModel, diagnostic, cancellationToken);
+                Contract.ThrowIfNull(arrayOp.Initializer);
+
+                var stringValue = GetUtf8StringValueFromArrayInitializer(arrayOp.Initializer);
+
+                // If our array is parented by a conversion to ReadOnlySpan<byte> then we don't want to call
+                // ToArray after the string literal, or we'll be regressing perf.
+                var isConvertedToReadOnlySpan = arrayOp.Parent is IConversionOperation conversion &&
+                    conversion.Type is INamedTypeSymbol { IsGenericType: true } namedType &&
+                    namedType.ConstructedFrom.Equals(readOnlySpanType) &&
```

Comment:
I'm a bit confused why this is ConstructedFrom, not OriginalDefinition. What's the difference between the two actually?

Reply:
No idea!

Reply:
Looks like OriginalDefinition.Equals with no comparer passed in is the most popular..